### DISCUSSION TITLE
To check the number of lines available from the current position to end of page

### DISCUSSION THREAD

**Discussion by ramkumarkb** [typst/typst#1863] [Q&A] [answered]
Many times we would like to test the remaining lines in the page before deciding whether to continue to place the content or insert a page-break and insert the content in the next page. This is usually the case for dynanic content.

`ConTeXt` has this neat feature - https://wiki.contextgarden.net/Command/testpage 

Is there a way to do something similar in `Typst` ?

**Comment by laurmaedje** [comment, marked as answer]
No that doesn't exist currently, you could open a feature request. 

While you can get a rough indication with `locate`, it will break pretty quickly if you shift content based on locate because it doesn't propagate through the whole document within Typst's five layout attempts.

**Reply by ramkumarkb** [reply]
@laurmaedje - Ok, noted.  I have raised a `feature-request` - https://github.com/typst/typst/issues/1864
